File: scripts/init-db.js
const sqlite3 = require('sqlite3').verbose();
const fs = require('fs');
const path = require('path');

const DB_PATH = process.env.DB_PATH || path.resolve(__dirname, '../db.sqlite');
const SCHEMA_PATH = process.env.SCHEMA_PATH || path.resolve(__dirname, '../schema.sql');

async function initializeDatabase() {
    try {
        const schema = fs.readFileSync(SCHEMA_PATH, 'utf8');
        const db = new sqlite3.Database(DB_PATH);

        await new Promise((resolve, reject) => {
            db.exec(schema, (err) => {
                if (err) {
                    console.error('Error executing schema:', err.message);
                    reject(err);
                } else {
                    console.log('Database initialized successfully.');
                    resolve();
                }
            });
        });

        db.close();
    } catch (error) {
        console.error('Failed to initialize database:', error.message);
        process.exit(1);
    }
}

if (require.main === module) {
    initializeDatabase();
}

File: scripts/init_db.py
import sys
import os
from pathlib import Path

# Add the parent directory of src to the Python path
sys.path.append(str(Path(__file__).resolve().parent.parent))

from src.database import connect_db

if __name__ == "__main__":
    print("Initializing database...")
    try:
        db_connection = connect_db()
        if db_connection:
            print("Database initialized successfully.")
            db_connection.close()
        else:
            print("Failed to initialize database: Connection not established.")
            sys.exit(1)
    except Exception as e:
        print(f"An error occurred during database initialization: {e}")
        sys.exit(1)
